package.json

```diff
@@ -31,6 +31,7 @@
     "@angular/core": "^4.0.0",
     "@angular/forms": "^4.0.0",
     "@angular/http": "^4.0.0",
+    "@angular/router": "^4.0.0",
```

Comment:
This doesn't introduce a dependency on the router, because we don't import the `RouterModule` anywhere and the injected `Router` is optional (it works in a similar way to the icons and the `HttpModule`). The difference between this and the `Location` is that `Location` only fires on hash changes and the user presses the back button, whereas the `Router` will also fire when clicking on links.